html/modules/custom/reliefweb_analytics/src/Command/ReliefwebMostReadCommand.php

```diff
@@ -0,0 +1,468 @@
+      '@h' => $quota->getTokensPerHour()->getConsumed() . '/' . $quota->getTokensPerHour()->getRemaining(),
+      '@e' => $quota->getServerErrorsPerProjectPerHour()->getConsumed() . '/' . $quota->getServerErrorsPerProjectPerHour()->getRemaining(),
+      '@t' => $quota->getPotentiallyThresholdedRequestsPerHour()->getConsumed() . '/' . $quota->getPotentiallyThresholdedRequestsPerHour()->getRemaining(),
+    ]));
+
+    foreach ($response->getRows() as $row) {
+      $results[] = $row->getDimensionValues()[0]->getValue();
+    }
+
+    return $results;
+  }
+
+  /**
+   * Fetch and process GA4 data.
+   *
+   * @param array $parameters
+   *   Payload.
+   *
+   * @see https://developers.google.com/analytics/devguides/reporting/core/v4/limits-quotas#analytics_reporting_api_v4
+   */
+  public function fetchGa4DataCombined(array $parameters) {
+    $results = [];
+
+    try {
+      $start = microtime(TRUE);
+      $response = $this->getGa4Client()->runReport($parameters);
+
+      // Make sure it takes at least a second.
+      $end = microtime(TRUE);
+      if ($end - $start < 1) {
+        usleep($end - $start);
+      }
+    }
+    catch (ApiException $exception) {
+      if ($exception->getStatus() == 'RESOURCE_EXHAUSTED') {
+        $this->logger()->warning('Rate limit hit.');
+        $this->logger()->warning('Google exception: ' . $exception->getMessage());
+      }
+      else {
+        $this->logger()->error('Google exception: ' . $exception->getMessage());
+      }
+      exit();
+    }
+    catch (\Exception $exception) {
+      $this->logger()->error('Exception: ' . $exception->getMessage());
+      exit();
+    }
+
+    // Log quota.
+    $quota = $response->getPropertyQuota();
+    $this->logger()->notice(strtr('Day: @d, hour: @h, errors/hour: @e, threshold: @t', [
+      '@d' => $quota->getTokensPerDay()->getConsumed() . '/' . $quota->getTokensPerDay()->getRemaining(),
+      '@h' => $quota->getTokensPerHour()->getConsumed() . '/' . $quota->getTokensPerHour()->getRemaining(),
+      '@e' => $quota->getServerErrorsPerProjectPerHour()->getConsumed() . '/' . $quota->getServerErrorsPerProjectPerHour()->getRemaining(),
+      '@t' => $quota->getPotentiallyThresholdedRequestsPerHour()->getConsumed() . '/' . $quota->getPotentiallyThresholdedRequestsPerHour()->getRemaining(),
+    ]));
+
+    foreach ($response->getRows() as $row) {
+      if (isset($row->getDimensionValues()[1]) && !empty($row->getDimensionValues()[1]->getValue()) && $row->getDimensionValues()[1]->getValue() !== '(not set)') {
+        if (!isset($results[$row->getDimensionValues()[1]->getValue()])) {
+          $results[$row->getDimensionValues()[1]->getValue()] = [];
+        }
+        if (count($results[$row->getDimensionValues()[1]->getValue()]) < 5) {
+          $results[$row->getDimensionValues()[1]->getValue()][] = $row->getDimensionValues()[0]->getValue();
+        }
```

Comment:
I'm not sure this will produce the expected result because some reports can be tagged with multiple disasters in that case `$row->getDimensionValues()[1]` may be a comma separated list of term labels (if using `customEvent:content_report_disaster`) or of term Ids (if using `dimension24` once available).

Reply:
The dimensions from `reliefweb_analytics_page_attachments` should be expanded again so not `dimension9 = 1,2,3` but `dimension9=1`, `dimension9=2`, `dimension9=3` which is supported by GA

Reply:
I stand corrected, dimensions are either number or string, so I'll expand in code